submissions/madmaxWMFU/Friends-App/js/app.js

```diff
@@ -0,0 +1,153 @@
+                </div>`;
+    friendZone.insertAdjacentHTML("beforeEnd", temp);
+}
+
+const cleanFriendsZone = () => friendZone.innerHTML = "";
+
+const showErrorMessage = () => friendZone.insertAdjacentHTML("beforeEnd", '<h4 class="friends-zone__error">Something went wrong...</h4>');
+
+const renderFriendsList = (allFriends) => allFriends.forEach(drawFriendsCards);
+
+const getFilterList = {
+    filterByGender: (list, {gender}) => {
+        if(gender !== "all") {
+          return list.filter((person) => person.gender === gender); 
+        } else {
+            return allFriends.currentList;
+        }
+    },
+    filterByAge: (list, {ageTill, ageTo}) => {
+        return list.filter((person) => ageTill <= person.dob.age && person.dob.age <= ageTo);
+    },
+    filterByName: (list, {name}) => {
+        if(name != "") {
+            return list.filter((person) => person.name.first.toLowerCase().includes(name));
+        } else {
+            return allFriends.changeList;
+        }
+    }
+}
+
+const getGenderValue = () => searchGender.find(item => item.checked === true).value;
+
+const applyFilters = () => {
+    Object.keys(getFilterList).forEach((filter) => {
+        allFriends.changeList = getFilterList[filter](allFriends.changeList, searchValues);
+    });
+}
+
+const renderFilterList = () => {
+    sorts.value = "none";
+    allFriends.changeList = [...allFriends.currentList];
+    searchValues.name = searchName.value;
+    searchValues.gender = getGenderValue();
+    searchValues.ageTill = searchAgeTill.value;
+    searchValues.ageTo = searchAgeTo.value;
+   
+    applyFilters();
+    cleanFriendsZone();
+    renderFriendsList(allFriends.changeList);  
+}
+
+const sortByName = (a, b) => a.name.first < b.name.first ? -1 : 1;
+
+const getSortByNameList = (option) => {
+    return allFriends.changeList.sort((firstPerson, secondPerson) => {
+        if(option === "nameAsc") {
+            return sortByName(firstPerson, secondPerson);
+        } else {
+            return sortByName(secondPerson, firstPerson);
+        }
+    });
+} 
+
+const sortByAge = (a, b) => a.dob.age - b.dob.age;
+
+const getSortByAgeList = (option) => {
+    return allFriends.changeList.sort((firstPerson, secondPerson) => {
+        if(option === "ageAsc") {
+            return sortByAge(firstPerson, secondPerson); 
+        } else {
+            return sortByAge(secondPerson, firstPerson);
+        }
+    });
+}
+
+const renderSortFriendList = ({target}) => {
+    if(target.value != "none")
+        allFriends.changeList = target.value.includes("name") ? getSortByNameList(target.value) : getSortByAgeList(target.value);
+        cleanFriendsZone();
+        renderFriendsList(allFriends.changeList);
+}
+
+const resetSearchValues = () => {
+    sorts.value = "none";
+    searchName.value = "";
+    searchAgeTill.value = allFriends.minAge;
+    searchAgeTo.value = allFriends.maxAge;
+    searchGender.find(item => item.value === "all").checked = true;
```

Comment:
As I said before, such constructions can be hard to read.
Please, can you describe the reason, why you prefer such approach instead of variable?

Reply:
Probably better to do so:
```javascript 
const allGender = document.querySelector('#all[name="search-input-gender"]');
or
const allGender = document.querySelector('#all');
```
Am I right?

Reply:
I meant something like this, actually :)

```javascript
const selectAllGenders = () => {
  const allGendersItem = selectedGenders.find(item => item.value === "all");

  if (allGendersItem) {
    allGendersItem.checked = true;
  }
};

selectAllGenders();
```